### DISCUSSION TITLE
Run in home assistant via pyscript

### DISCUSSION THREAD

**Discussion by silviudc** [ZuinigeRijder/SolisCloud2PVOutput#13] [Q&A]
Anyone got this to run in Home Assistant using something like pyscript in HACS?
https://hacs-pyscript.readthedocs.io/en/latest/
I tried hacking this code together but only thing I manage to do is crash the HA server, python is not my thing it seems

**Comment by ZuinigeRijder** [comment]
In the end, it is the lines  206-210 which contains the information to send to PVOutput (comma separated line):

````
            pvoutput_string = (
                'data=' + TODAY +  # Date
                ',' + datetime_current.strftime("%H:%M") +  # Time
                ',' + str(hi_res_watthour_today) +  # Energy Generation
                ',' + str(watt) +  # Power Generation
                ',-1' +  # no Energy Consumption
                ',' + str(ac_volt) +  # Power generation used for AC voltage
                ',' + str(inverter_temp) +  # inverter temp iso outside temp
                ',' + str(volt)  # Voltage
            )
````

So you should be able to store this somehow in HACS. But I have no experience with HACS and also do not have a HA system.

**Comment by ZuinigeRijder** [comment]
I forget to mention, there is a solis sensor for HACS.

https://github.com/hultenvp/solis-sensor
